pkg/network/kuryr.go

```diff
@@ -113,7 +113,15 @@ func renderKuryr(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.BootstrapR
 	data.Data["WebhookKey"] = b.WebhookKey
 
 	// Nodes Network MTU
-	data.Data["NodesNetworkMTU"] = b.NodesNetworkMTU
+	mtu := b.NodesNetworkMTU
+	if c.MTU != nil && *c.MTU != 0 {
+		if mtu >= *c.MTU {
+			mtu = *c.MTU
+		} else {
+			return nil, errors.Errorf("Provided MTU (%d) is too big.", *c.MTU)
+		}
+	}
+	data.Data["NodesNetworkMTU"] = mtu
```

Comment:
We could probably rename `NodesNetworkMTU` to `PodNetworkMTU`, but that's not a big deal.